angular-chart.js

```
@@ -1,243 +1,302 @@
-        var chart, container = document.createElement('div');
-        container.className = 'chart-container';
-        elem.replaceWith(container);
-        container.appendChild(elem[0]);
-
-        if (typeof window.G_vmlCanvasManager === 'object' && window.G_vmlCanvasManager !== null) {
-          if (typeof window.G_vmlCanvasManager.initElement === 'function') {
-            window.G_vmlCanvasManager.initElement(elem[0]);
-          }
-        }
+    .provider('ChartJs', ChartJsProvider)
+    .factory('ChartJsFactory', ['ChartJs', ChartJsFactory])
+    .directive('chartBase',
+      ['ChartJsFactory', function (ChartJsFactory) { return new ChartJsFactory(); }])
+    .directive('chartLine',
+      ['ChartJsFactory', function (ChartJsFactory) { return new ChartJsFactory('Line'); }])
+    .directive('chartBar',
+      ['ChartJsFactory', function (ChartJsFactory) { return new ChartJsFactory('Bar'); }])
+    .directive('chartRadar',
+      ['ChartJsFactory', function (ChartJsFactory) { return new ChartJsFactory('Radar'); }])
+    .directive('chartDoughnut',
+      ['ChartJsFactory', function (ChartJsFactory) { return new ChartJsFactory('Doughnut'); }])
+    .directive('chartPie',
+      ['ChartJsFactory', function (ChartJsFactory) { return new ChartJsFactory('Pie'); }])
+    .directive('chartPolarArea',
+      ['ChartJsFactory', function (ChartJsFactory) { return new ChartJsFactory('PolarArea'); }]);
+
+  /**
+   * Wrapper for chart.js
+   * Allows configuring chart js using the provider
+   *
+   * angular.module('myModule', ['chart.js']).config(function(ChartJsProvider) {
+   *   ChartJsProvider.setOptions({ responsive: true });
+   *   ChartJsProvider.setOptions('Line', { responsive: false });
+   * })))
+   */
+  function ChartJsProvider() {
+    var Chart = (typeof window !== 'undefined' && window.Chart) ||
+      (typeof require !== 'undefined' && require('chart.js'));
+
+    if (typeof Chart !== 'function') {
+      throw 'Failed to load chart.js';
```

Need to throw an error i.e. `new Error('Failed to load chart.js')`.